config/initializers/data_providers.rb

```diff
@@ -0,0 +1,16 @@
+providers_config = YAML.safe_load(ERB.new(File.read(Rails.root.join("config", "data_providers.yml"))).result)
+
+class ProviderFactory
+    def self.create(name, api_key)
+        "DataProvider::#{name}".constantize.new(api_key)
```

Comment:
Better to just index into an array of permitted values for this and avoid `constantize`

See:
http://gavinmiller.io/2016/the-safesty-way-to-constantize/